uwblueprint/supportive-housing
# Invite Employees modal

## GitHub Issue link

[Invite Employee Form](https://github.com/uwblueprint/supportive-housing/issues/94)



## Implementation description
*  a modal to allow admins to invite an employee to the system
* once the form is submitted, the employee is correctly invited and a user is created in the DB

![Screenshot 2023-07-06 at 11 25 21 PM](https://github.com/uwblueprint/supportive-housing/assets/87948268/068cbc97-8afc-4877-985e-03dc0523c5b5)


## Steps to test
1. Make sure you are logged in as an admin
2. Go to http://localhost:3000/invite-employees
3. Open the modal
4. Note that these inputs are required: `First Name, Last Name, Admin Status`, so play around with it and see if errors pop up when you try to submit but some or all of those fields are empty
![Screenshot 2023-07-06 at 11 32 55 PM](https://github.com/uwblueprint/supportive-housing/assets/87948268/8cdaa951-91dd-40fc-a9b5-226ca4b408c8)

5. make sure that the email input field must be a SHOW email, and a warning pops up if not a SHOW email
6. make sure that when the `admin radio` is selected, the checkbox is disabled
7. Now try actually inviting the users and check that it is in the users db

- when you select: admin, the user role is admin
- when you select non-admin + needs 2Fa, the user role is relief staff
- when you select non-admin + doesn't need 2fa, the user role is regular staff

8. Make sure the success toast shows up
![Screenshot 2023-07-07 at 11 31 26 AM](https://github.com/uwblueprint/supportive-housing/assets/87948268/0a492a9e-394f-4cb6-bd92-cb77b79f06c0)

10. Also do some hardcode to see if the error toast shows up correctly



## What should reviewers focus on?
* the user roles are right 
* toasts
* Does the font difference between the input boxes and the radios + checkbox font matter




## Checklist
